Jenkinsfile

```diff
@@ -16,293 +16,345 @@ pipeline {
+                          -v "$WORKSPACE/bundles:/go/src/github.com/docker/docker/bundles" \
+                          -v "$WORKSPACE/.git:/go/src/github.com/docker/docker/.git" \
+                          --name docker-pr$BUILD_NUMBER \
+                          -e DOCKER_GITCOMMIT=${GITCOMMIT} \
+                          -e DOCKER_GRAPHDRIVER=overlay2 \
+                          docker:$GITCOMMIT \
+                          hack/test/unit
+                        '''
+                    }
+                    post {
+                        always {
+                            junit 'bundles/junit-report.xml'
+                            sh label: 'Remove dev container', script: '''
+                            echo 'Ensuring container killed.'
+                            docker rm -vf docker-pr$BUILD_NUMBER || true
+                            '''
 
-                docker run --rm -t --privileged \
-                  -v "$WORKSPACE/bundles:/go/src/github.com/docker/docker/bundles" \
-                  -v "$WORKSPACE/.git:/go/src/github.com/docker/docker/.git" \
-                  --name docker-pr$BUILD_NUMBER \
-                  -e DOCKER_GITCOMMIT=${GITCOMMIT} \
-                  -e DOCKER_GRAPHDRIVER=overlay2 \
-                  docker:$GITCOMMIT \
-                  hack/test/unit
-              '''
-          }
-          post {
-            always {
-              junit 'bundles/junit-report.xml'
-              sh '''
-                echo 'Ensuring container killed.'
-                docker rm -vf docker-pr$BUILD_NUMBER || true
+                            sh label: 'Chown workspace to jenkins user', script: '''
+                            echo 'Chowning /workspace to jenkins user'
+                            docker run --rm -v "$WORKSPACE:/workspace" busybox chown -R "$(id -u):$(id -g)" /workspace
+                            '''
 
-                echo 'Chowning /workspace to jenkins user'
-                docker run --rm -v "$WORKSPACE:/workspace" busybox chown -R "$(id -u):$(id -g)" /workspace
+                            sh label: 'Creating artifacts', script: '''
+                            echo 'Creating unit-bundles.tar.gz'
+                            tar -czvf unit-bundles.tar.gz bundles/junit-report.xml bundles/go-test-report.json bundles/profile.out
+                            '''
 
-                echo 'Creating unit-bundles.tar.gz'
-                tar -czvf unit-bundles.tar.gz bundles/junit-report.xml bundles/go-test-report.json bundles/profile.out
-              '''
-              archiveArtifacts artifacts:'unit-bundles.tar.gz'
-              deleteDir()
-            }
-          }
-        }
-        stage('janky') {
-          when {
-            beforeAgent true
-            expression { params.janky }
-          }
-          agent { label 'amd64 && ubuntu-1804 && overlay2' }
-          environment { DOCKER_BUILDKIT='1' }
+                            sh label: 'Make clean', script: '''
```

Comment:
Should this be in a `cleanup` block in the Jenkinsfile? https://jenkins.io/doc/book/pipeline/syntax/#post

Reply:
Rewriting the PR now to use substages; good suggestion, I'll use a `cleanup` block for this